src/app_modes/teleoperate.rs

```diff
@@ -63,11 +77,24 @@ impl AppMode for Teleoperate {
     }
 
     fn run(&mut self) {
-        let mut vel_cmd = rosrust_msg::geometry_msgs::Twist::default();
-        vel_cmd.linear.x = self.current_velocities.x;
-        vel_cmd.linear.y = self.current_velocities.y;
-        vel_cmd.angular.z = self.current_velocities.theta;
-        self.cmd_vel_pub.send(vel_cmd).unwrap();
+        // If the velocity is reset to 0 only publish it once
+        // this prevents the robot from being blocked if the
+        // app mode is not closed
+        if  self.force_stop_when_idle
```

Comment:
i would prefer to not only do it for 0 velocities but for all velocities, ie: send a zero command if no button was pressed for x seconds (5?) and stop publishing

Reply:
Not so sure here, I think most implementations would assume that the machine should stop.